stanza/routines/builtins/dqd_search/routine.py

```diff
@@ -0,0 +1,692 @@
+import logging
+import uuid
+from typing import Any
+
+import numpy as np
+
+from stanza.logger.session import LoggerSession
+from stanza.routines.builtins.dqd_search.grid_search import (
+    GRID_SQUARE_MULTIPLIER,
+    SearchSquare,
+    generate_2d_sweep,
+    generate_diagonal_sweep,
+    generate_grid_corners,
+    select_next_square,
+)
+from stanza.routines.builtins.dqd_search.utils import (
+    build_full_voltages,
+    build_gate_indices,
+    check_voltages_in_bounds,
+    compute_peak_spacings,
+    generate_linear_sweep,
+    generate_random_sweep,
+    get_gate_safe_bounds,
+    get_global_turn_on_voltage,
+    get_plunger_gate_bounds,
+    get_voltages,
+    log_classification,
+    log_measurement,
+    measure_and_classify,
+)
+from stanza.routines.core import RoutineContext, routine
+
+logger = logging.getLogger(__name__)
+
+
+@routine
+def compute_peak_spacing(
+    ctx: RoutineContext,
+    gates: list[str],
+    measure_electrode: str,
+    min_peak_scale: float,
```

Comment:
I'd maybe use something like min_search_scale, since it's not necessarily the peak scale